File: Byte&Buy.AngularSPA/src/app/core/dto/order/order-details-response.ts
import { Guid } from "guid-typescript";
import { OrderStatus } from "./enum/order-status";
import { MoneyDto } from "../common/money-dto";
import { CourierDeliveryDetails, ParcelLockerDeliveryDetails, PickupPointDeliveryDetails } from "../order-delivery/order-delivery-details";
import { HomeAddressDto } from "../home-address/home-address-dto";
import { UserOrderLineResponseUnion } from "./common/user-order-list-response";

export type OrderDetailsUnion = 
    | CourierDeliveryDetails
    | PickupPointDeliveryDetails
    | ParcelLockerDeliveryDetails

export interface OrderDetailsResponse {
  id: Guid; 
  status: OrderStatus;
  purchasedDate: Date; 
  dateDelivered?: string | null;
  sellerDisplayName: string;
  linesCount: number;
  totalItemsCost: MoneyDto;
  totalCost: MoneyDto;
  deliveryDetails: OrderDetailsUnion;
  buyerSnapshot: BuyerSnapshotResponse;
  lines: UserOrderLineResponseUnion[];
}

export interface BuyerSnapshotResponse {
  fullName: string;
  email: string;
  phoneNumber: string;
  address: HomeAddressDto;
}

File: Byte&Buy.Desktop/src/ByteBuy.Services/DTO/Order/OrderDetailsResponse.cs
using ByteBuy.Services.DTO.Address;
using ByteBuy.Services.DTO.Money;
using ByteBuy.Services.DTO.Order.Enums;
using ByteBuy.Services.DTO.Order.OrderLine;
using ByteBuy.Services.DTO.OrderDelivery;

namespace ByteBuy.Services.DTO.Order;
public sealed record BuyerSnapshotResponse(
     string FullName,
     string Email,
     string PhoneNumber,
     HomeAddressDto Address);

public sealed record OrderDetailsResponse(
    Guid Id,
    Guid? PaymentId,
    OrderStatus Status,
    DateTime PurchasedDate,
    DateTime? DateDelivered,
    string SellerDisplayName,
    int LinesCount,
    MoneyDto TotalItemsCost,
    MoneyDto TotalCost,
    OrderDeliveryDetails DeliveryDetails,
    BuyerSnapshotResponse BuyerSnapshot,
    IReadOnlyCollection<UserOrderLineResponse> Lines);
